app/scripts/controllers/taskController.js

```diff
@@ -0,0 +1,17 @@
+(function() {
+    function taskController (Tasks, $scope, $interval, $firebaseArray) {
+      this.all = Tasks.all
+      var ref = firebase.database().ref().child("tasks");
```

Comment:
I don't think you need lines 4 and 5 since the tasks service handles all communication with Firebase, which also means you can remove $firebaseArray as a dependency for this controller!